### PR TITLE
조건문을 private한 메서드로 만들고 widget으로 빼보았습니다.

### PR DESCRIPTION
**Pull request by chepchep2**
RegisterPage에서 각 조건문을 private한 메서드로 빼고 각 로직들을 widget으로 만들었습니다.이렇게 하는게 build메서드에서 조금 더 가독성이 좋아보여서 구분지어봤습니다.

### CODE REVIEW

**Comment by chepchep2** [comment]
* **#64** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/64?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#65](https://github.com/team-stoodi/MyBaseballRecordClient/pull/65) <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/65?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#66](https://github.com/team-stoodi/MyBaseballRecordClient/pull/66) <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/66?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#63** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/63?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#62** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/62?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#61** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/61?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#60** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/60?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#59** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/59?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#58** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/58?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#57** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/57?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#56** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/56?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#55** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/55?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#54** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/54?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#53** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/53?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#52** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/52?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#51** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/51?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#50** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/50?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#49** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/49?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#48** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/48?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#47** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/47?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#46** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/46?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#45** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/45?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#44** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/44?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#43** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/43?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#42** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/42?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#41** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/41?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#40** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/40?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#39** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/39?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/35?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#34** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/34?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `feature/introPage`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @chepchep2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/page/register_page.dart [hunk 1]
```diff
@@ -62,6 +62,117 @@ class _RegisterPageState extends State<RegisterPage> {
...
+        style: AppTextStyle.caption213R.copyWith(
+          color: AppColor.textHint,
+        ),
+      );
+    }
+    return Container();
+  }
+
+  Widget _buildPasswordErrorText() {
+    if (_passwordError.isNotEmpty) {
```
Comment (flowkater) [inline lib/page/register_page.dart:117]: if 문이 위젯 안에 들어가는 건 별로 안좋은 것 같고, 굳이 Text 위젯을 이렇게까지 잘게 나눌 필요는 없을 것 같습니다.

아니면 한가지 방법은
_buildEmailInfoText 로 묶어서 Error 랑 정보 Text 랑 묶어서 노출을 하는 것도 한 방법일 것 같습니다.
email, password, confirmPassword 이렇게

함수를 나눌때 너무 잘게 쪼개는 것보다 목적별로 나누는게 좋을 것 같습니다.

그리고 Container() 보다 그냥 null 을 반환하면 에러가 나나요?

```dart
Widget _buildEmailErrorText() {
  return Visibility(
    visible: _emailError.isNotEmpty,
    child: Text(
      _emailError,
      style: AppTextStyle.caption213R.copyWith(
        color: AppColor.accentRed100,
      ),
    ),
  );
}
```

이런 식도 가능할 것 같고,
삼항 연산자로 null 을 반환하거나 SizedBox.shrink() 를 하는 것도 한 방법일 것 같습니다.

**Review by flowkater — CHANGES_REQUESTED** [review]
위 리뷰 참고부탁드립니다!

**Comment by chepchep2** [comment]
### Merge activity

* **Apr 7, 10:24 AM EDT**: @chepchep2 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/62).
* **Apr 7, 12:00 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/62) rebased this pull request as part of a merge.
* **Apr 7, 12:01 PM EDT**: @chepchep2 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/62).

**Pull request merged**